bin/ubuntu-core-initramfs

```diff
@@ -616,6 +611,58 @@ def create_initrd_pkg_list(dest_dir, sysroot):
                             pkgs).decode("utf-8")
         pkg_list.write(out)
 
+def verify_missing_dlopen(destdir, libdir):
+    missing = {}
+    for dirpath, dirs, files in os.walk(destdir):
+        for f in files:
+            path = os.path.join(dirpath, f)
+            if os.path.islink(path) or not os.path.isfile(path):
+                continue
+            with open(path, 'rb') as b:
+                if b.read(4) != b'\x7fELF':
+                    continue
+            out = check_output(["dlopen-notes", path])
+            split = out.splitlines()
+            json_doc = b'\n'.join([s for s in split if not s[:1] == b'#'])
+            doc = json.loads(json_doc)
+            for dep in doc:
+                sonames = dep["soname"]
+                priority = dep["priority"]
+                found_sonames = []
+                for soname in sonames:
```

Comment:
Is this a list of alternative implementation of the feature?

Reply:
No, it is a list of shared object names. That is it might try to dlopen `libfoo.so.1`, and if that does not work `libfoo.so.2`. In that case the sonames will be a list of 2 strings.

Reply:
Both things are true, aren't they? :slightly_smiling_face: